### ISSUE TITLE
Popover: can't make it fit inside a mobile screen

### ISSUE THREAD

**Issue by KKoukiou** [patternfly/patternfly#4754]
I have trouble making the popover component adjust it's width to fit inside a mobile screen. That said passing `hasAutoWidth` or not, the popovers always make it out of the screen like seen in the sceenshot bellow.
![Screen Shot 2021-07-07 at 18 13 21](https://user-images.githubusercontent.com/14921356/124794319-2b7db080-df4f-11eb-9e0b-e97471eba272.png)

**Comment by mcarrano** [comment]
@garrett @KKoukiou we will get this on our roadmap to take a deeper look at updating the default behavior.

**Comment by blfetche** [comment]
Seems there is an option within Popper JS that can act as a possible solution to this issue. Big thanks to @mcoker for this find. I have implemented it into my sandbox. Seems to work well regardless of the trigger placement.

https://codesandbox.io/s/popover-auto-width-forked-swvie9?resolutionWidth=579&resolutionHeight=675&file=/index.js

**Comment by blfetche** [comment]
@KKoukiou @garrett does the sandbox solution help?

**Comment by garrett** [comment]
@blfetche: This does look good, even in small widths. I checked it in Firefox, Chrome, and GNOME Web (all the browser engines). Screenshot from GNOME Web (aka: Epiphany, WebKit), but everything else looks the same too:

![image](https://user-images.githubusercontent.com/10246/184871424-6aa5e5f7-14da-4a59-917e-afb50f4b1a20.png)

The "magic" looks like a combination of using `hasAutoWidth` and _especially_ specifying all the different directions with `flipBehavior`? And I see it's at the top of the document... I wonder if that would make a difference as well? (If it's deep in the DOM, I wonder if it still works, in other words.)

**Comment by blfetche** [comment]
@garrett thanks for you feedback, if I understand your saying, if its placed in other areas on screen, how would it behave?

**Comment by garrett** [comment]
@blfetche: Yes, definitely. But also: Do we need to have all of that every time we use a tooltip? Or can PF be made to handle it by default?

**Comment by blfetche** [comment]
Updated the sandbox environment to include a layout example to show varying placements for the popper as a test. Below are images showing results of auto placement. 

<img width="746" alt="image" src="https://user-images.githubusercontent.com/96086344/185526110-78a12c81-a1ef-4def-a2b4-91c3db84837d.png">

<img width="747" alt="image" src="https://user-images.githubusercontent.com/96086344/185526558-f2039b17-c7d3-4911-8f04-2c8b90cfd09e.png">

<img width="799" alt="image" src="https://user-images.githubusercontent.com/96086344/185528058-da061e23-faf8-42d5-8422-6bef421daae7.png">

**Comment by garrett** [comment]
Can we have this set so it's default? It's not ideal to write so much custom JSX or HTML for each and every popover.

**Comment by blfetche** [comment]
@garrett going to tie this to a React issue that could offer a default solution. Also think the optimal solution is diagonal placement as there are still cases where the popper gets cropped given the most recent sandbox examples, specifically when it is centered above/below an element.

Updated Issue: https://github.com/patternfly/patternfly-react/issues/7762

**Comment by mcarrano** [comment]
Thanks @blfetche . 

@garrett I am going to close this in favor of the React issue above that hopefully should address your concern. This is queued up for a future milestone.

**Comment by blfetche** [comment]
Thanks @mcarrano
